src/components/Countdown/Countdown.tsx

```diff
@@ -0,0 +1,151 @@
+import { ComponentProps, forwardRef, useCallback, useEffect, useState } from 'react';
+import { getCommonProps } from '../../utils';
+import classnames from 'classnames';
+
+// You'll need to change the ComponentProps<"htmlelementname"> to match the top-level element of your component
+export interface CountdownProps extends ComponentProps<'div'> {
+  /**
+   * The date the countdown should end
+   */
+  endDate: string;
+  /**
+   * Any descriptor to appear below the coundown
+   */
+  intervalDescription?: string;
+  /**
+   * Text translation for days
+   */
+  daysLabel?: string;
+  /**
+   * Text translation for hours
+   */
+  hoursLabel?: string;
```

Comment:
if you take the locale as a parameter, you could just use this I believe to format the words for Hours, Minutes, Days, and Seconds: https://date-fns.org/v4.1.0/docs/formatDistanceStrict